packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -556,11 +556,19 @@ class AnimationController extends Animation<double>
     Duration? simulationDuration = duration;
     if (simulationDuration == null) {
       assert(() {
-        if ((this.duration == null && _direction == _AnimationDirection.reverse && reverseDuration == null) || this.duration == null) {
+        if (this.duration == null && _direction == _AnimationDirection.reverse && reverseDuration == null) {
```

Comment:
Instead of throwing error in this private method, can we throw in animateTo and animateBack separately, and convert the throw here to an assertion only?

Reply:
Thanks for your review suggestion. I will revise it ^o^